File: src/main/java/org/taonaw/reservation/domain/model/reservations/RentalEquipment.java
package org.taonaw.reservation.domain.model.reservations;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NonNull;
import org.taonaw.common.Assertion;
import org.taonaw.reservation.domain.model.equipments.EquipmentId;

@Getter
@EqualsAndHashCode
public class RentalEquipment {
    private final EquipmentId equipmentId;
    private final int quantity;

    public RentalEquipment(@NonNull EquipmentId equipmentId, int quantity) {
        Assertion.argumentMin(quantity, 1);
        this.equipmentId = equipmentId;
        this.quantity = quantity;
    }

    public RentalEquipment increase(int quantity) {
        return new RentalEquipment(this.equipmentId, this.quantity + quantity);
    }
}

File: src/backend/reservation/src/main/kotlin/org/taonaw/reservation/domain/model/reservation/rentalEquipment/RentalEquipment.kt
package org.taonaw.reservation.domain.model.reservation.rentalEquipment

import org.taonaw.reservation.domain.model.equipment.EquipmentId

/**
 * レンタル機材
 */
class RentalEquipment(
        val equipmentId: EquipmentId,
        val quantity: RentalEquipmentQuantity) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as RentalEquipment

        if (equipmentId != other.equipmentId) return false

        return true
    }

    override fun hashCode(): Int {
        return equipmentId.hashCode()
    }
}